specification/snapshot/_types/SnapshotRepository.ts

```diff
@@ -17,22 +17,132 @@
  * under the License.
  */
 
-import { Uuid } from '@_types/common'
+import { ByteSize, Uuid } from '@_types/common'
 import { integer } from '@_types/Numeric'
+import { Base } from '@xpack/usage/types'
+import { AdditionalProperties } from '@spec_utils/behaviors'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { Stringified } from '@spec_utils/Stringified'
+import { Duration } from '@_types/Time'
 
-export class Repository {
-  type: string
+export class BaseRepository {
   uuid?: Uuid
-  settings: RepositorySettings
 }
 
-export class RepositorySettings {
-  chunk_size?: string
-  compress?: string | boolean
-  concurrent_streams?: string | integer
+/**
+ * @variants internal tag='type'
+ * @non_exhaustive
+ */
+export type Repository =
+  | S3Repository
+  | GoogleCloudRepository
+  | AzureRepository
+  | SourceRepository
+  | UrlRepository
+  | FsRepository
+  | HdfsRepository
+
+export class S3Repository extends BaseRepository {
+  type: 's3'
+  settings: S3RepositorySettings
+}
+
+class BlobStoreSettings {
+  chunk_size?: ByteSize
+  concurrent_streams?: Stringified<integer>
+  compress?: Stringified<boolean>
+  max_number_of_snapshots?: integer
+  max_snapshot_bytes_per_sec?: ByteSize
+  max_restore_bytes_per_sec?: ByteSize
+  /** @aliases read_only */
```

Comment:
Is there a reason we switched the `@aliases` and the non-alias property name from the previous types?

Reply:
This is because some recordings use `read_only` but the setting name in the ES code is `readonly`. I couldn't find though in the ES code where that `read_only` would be accepted.